server/service/service.go

```diff
@@ -53,12 +58,49 @@ func New(conf *config.Config) (*Service, error) {
 		}
 	}
 
+	var key []byte
+	if tos.FileExists(conf.DataEncryptionKeyFile) {
+		file, err := tos.ReadFile(conf.DataEncryptionKeyFile)
+		if err != nil {
+			return nil, err
+		}
+
+		key, err = ecrypto.Unseal(file, nil)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		key, err = crypto.GenerateRandomKey(32)
+		if err != nil {
+			return nil, err
+		}
+
+		sealed, err := ecrypto.SealWithProductKey(key, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// ex) $HOME/config/data_encryption_file.sealed
+		// dir = $HOME/config/, file = data_encryption_file.sealed
+		dir, _ := filepath.Split(conf.DataEncryptionKeyFile)
+		err = tos.EnsureDir(dir, 0755)
+		if err != nil {
+			return nil, err
+		}
+
+		err = tos.WriteFile(conf.DataEncryptionKeyFile, sealed, 0755)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Good. I recommend you to extract this part as a function which returns `[]byte, error`. If `err != nil`, you need to clean resources that have been initialized above (e.g. `panaceaClient.Close()`).

Reply:
I will extract this part as a function after this [PR](#73 ) is merged to this branch.

Reply:
Refacored in [2a8bdff](https://github.com/medibloc/panacea-data-market-validator/pull/71/commits/2a8bdff83a6e7c3a2dd76cc419386b626e665868).